std/file.d

```diff
@@ -4308,7 +4307,24 @@ string tempDir() @trusted
         }
         else static assert(false, "Unsupported platform");
 
-        if (cache is null) cache = getcwd();
+        if (cache is null)
+        {
+            import std.path : dirSeparator;
+            cache = getcwd() ~ dirSeparator;
+        }
     }
     return cache;
 }
+
+@trusted unittest
+{
+    version(Windows)
+    {
+        enum pathSeparator = '\\';
+    }
+    else
+    {
+        enum pathSeparator = '/';
```

Comment:
I'll change it tonight, but my thinking is that I want to force a failure if the meaning of the test changes, so a programmer has to make an explicit choice. I generally try to avoid pulling constant-ish values into tests for this reason.

If we were to add support for SomeNewOS that used, say, a caret as path separator, using `std.path.dirSeparator` in the test means that as long as `getcwd()` is implemented for SomeNewOS, the test will pass even if there's a temp directory we could be using. By forcing the test to fail, we are reminded to look for a temp directory for the OS and update this function accordingly.

In this specific case, this is a somewhat ludicrous scenario, as nobody is going to create a new OS with anything other than a slash as a path separator (at least for hierarchical filesystems) if they don't want to create headaches for themselves, so I'll fix the test.